### PR TITLE
Add COOL language

### PR DESCRIPTION
**Pull request by linonetwo**
Classroom Object Oriented Language is a widely used language in CS classes.

This Grammar defination comes from [COOL manual](http://sist.shanghaitech.edu.cn/faculty/songfu/course/spring2017/cs131/COOL/COOLAid.pdf).

This is a byproduct from a COOL-to-javascript compiler.

### CODE REVIEW

**Comment by KvanTTT** [comment]
Cool!

File: cool/COOL.g4 [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+WHITESPACE: [ \t\r\n\f]+ -> skip; 
+
+// comments
+OPEN_COMMENT: '(*';
+CLOSE_COMMENT: '*)';
+COMMENT: OPEN_COMMENT (COMMENT|.)*? CLOSE_COMMENT -> channel(HIDDEN);
+ONE_LINE_COMMENT: '--' .*? '\n' -> channel(HIDDEN);
+
+// key words
+CLASS: ('C'|'c')('L'|'l')('A'|'a')('S'|'s')('S'|'s');
```
Comment (KvanTTT) [inline cool/COOL.g4]: Can we use a more clear syntax here? I mean fragment rules: [Case insensitive keywords](http://blog.ptsecurity.com/2016/06/theory-and-practice-of-source-code.html#case-insensitive-keywords).

**Comment by teverett** [comment]
@KvanTTT I'm going to pull this, but I agree with your comment about clear syntax.  Could you submit a issue for it?

@linonetwo thanks very much for your contribution.  Also, thanks for submitting so many test cases; they will ensure that as the Antlr tree evolves, your grammar will continue to work.

**Comment by KvanTTT** [comment]
Ok, maybe later.

File: cool/COOL.g4 (hunk 1 above)
Comment (linonetwo) [inline cool/COOL.g4]: Sure, I will PR for that tomorrow morning. Thanks for that suggestion.

**Pull request merged**